File: CODIGOS_PYTHON/ACT1.py
class Padre:    #Clase Padre
    def __init__(self, nombre):
        self.nombre = nombre  #Atributo "nombre"
        self.hijos = []       #Lista para almacenar objetos Hijo
        self.numHijos = 0     #Contador de hijos

    #Metodo para agregar un hijo a la lista
    def agregarHijo(self, hijo):
        if self.numHijos < 7:
            self.hijos.append(hijo)
            self.numHijos += 1
        else:
            print("No se pueden agregar más hijos.")
    
    #Metodo para imprimir informacion del Padre
    def imprimirInfo(self):
        print("Nombre del Padre:", self.nombre)
        print("Número de Hijos:", self.numHijos)

#Clase Hijo
class Hijo:
    def __init__(self, nombre):
        self.nombre = nombre  # Atributo "nombre"

#Función principal
def main():
    padre = Padre("Alejandro")  #Crea un objeto Padre
    hijo1 = Hijo("Diego")  #Crea un objeto Hijo
    hijo2 = Hijo("Carmen")
    hijo3 = Hijo("Joaquin")
    hijo4 = Hijo("Mariana")

    padre.agregarHijo(hijo1)  #Agrega hijo1 al Padre
    padre.agregarHijo(hijo2)  #Agrega hijo2 al Padre
    padre.agregarHijo(hijo3)  #Agrega hijo2 al Padre
    padre.agregarHijo(hijo4)  #Agrega hijo2 al Padre

    padre.imprimirInfo()  #Imprime la información del Padre

if __name__ == "__main__":
    main()

File: CODIGOS_JAVA/ACT1.java
// Importa las clases necesarias para la entrada/salida
import java.util.ArrayList; // Importa la clase ArrayList para manejar colecciones dinámicas

// Define la clase Hijo
class Hijo {
    private String nombre; // Atributo privado "nombre" de tipo String

    // Constructor que inicializa el nombre del hijo
    public Hijo(String nombre) {
        this.nombre = nombre;
    }

    // Método get para obtener el nombre del hijo
    public String getNombre() {
        return nombre;
    }

    // Método set para establecer el nombre del hijo
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
}

// Define la clase Padre
class Padre {
    private String nombre; // Atributo privado "nombre" de tipo String
    private ArrayList<Hijo> hijos; // Lista para almacenar objetos Hijo

    // Constructor que inicializa el nombre y la lista de hijos
    public Padre(String nombre) {
        this.nombre = nombre;
        this.hijos = new ArrayList<>();
    }

    // Método get para obtener el nombre del padre
    public String getNombre() {
        return nombre;
    }

    // Método set para establecer el nombre del padre
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    // Método para agregar un hijo al ArrayList de hijos
    public void agregarHijo(Hijo hijo) {
        if (hijos.size() < 5) { // Comprueba si hay menos de 5 hijos antes de agregar uno nuevo
            hijos.add(hijo);
        } else {
            System.out.println("No se pueden agregar más hijos.");
        }
    }

    // Método para imprimir información del padre y sus hijos
    public void imprimirInfo() {
        System.out.println("Nombre del Padre: " + nombre);
        System.out.println("Número de Hijos: " + hijos.size());
    }
}

// Clase principal para ejecutar el programa
public class Act1 {
    public static void main(String[] args) {
        Padre padre = new Padre("Alexander"); // Crea un objeto Padre

        Hijo hijo1 = new Hijo("Pedro"); // Crea un objeto Hijo
        Hijo hijo2 = new Hijo("Maria"); // Crea otro objeto Hijo

        padre.agregarHijo(hijo1); // Agrega hijo1 al Padre
        padre.agregarHijo(hijo2); // Agrega hijo2 al Padre

        padre.imprimirInfo(); // Imprime la información del Padre
    }
}
